File: Examples/cosmology/xi0_Kaiser.py
## ==============================================================================
## Example code: how to compute the z-space matter monopole from the Kaiser model
## ==============================================================================

# to ensure compatibility in Python versions 2.x and 3.x
from __future__ import print_function

# import the CosmoBolognaLib and numpy modules
import CosmoBolognaLib as cbl
import numpy as np

# set the cosmological model, using default parameters
cosmology = cbl.LCDM("Planck18")

# create the object used for the computation of the matter power spectrum
PXS = cbl.PkXizSpace(cosmology)
print("xi0(r=1 Mpc/h, fsigma8=1, bsigma8=1, z=1) =", PXS.xi0_Kaiser(1., 1., 1., "EisensteinHu", 1.))

File: Examples/cosmology/xi0_Kaiser.cpp
// ==============================================================================
// Example code: how to compute the z-space matter monopole from the Kaiser model
// ==============================================================================

#include "LCDM.h"
#include "PkXizSpace.h"

using namespace std;

int main () {
    
  try {
    
    // --------------------------------------------------------
    // ---------------- set the cosmological model ------------
    // --------------------------------------------------------
    
    auto cosmology = make_shared<cbl::cosmology::LCDM>("Planck18");
    
    cosmology->set_parameter("sigma8",0.8111);
    
    
    // ----------------------------------------------------------------
    // ---------------- estimate the matter power spectrum ------------
    // ----------------------------------------------------------------
    
    cbl::cosmology::PkXizSpace PXS(cosmology);

    cout << "xi0(r=1 Mpc/h, fsigma8=1, bsigma8=1, z=1) = " << PXS.xi0_Kaiser(1., 1., 1., "EisensteinHu", 1.) << endl;
  }

  catch(cbl::glob::Exception &exc) { cerr << exc.what() << endl; exit(1); }

  return 0;
}
